File: 3/go/main.go
package main

func solve(n int) int {
	var i int = 2;

	for i * i <= n{
		println(n)
		if n % i != 0 {
			i++
		}else {
			n = n/i
		}

	}
	return n
}

func main(){
	println(solve(600851475143))
}

File: 3/python/main.py
def solve(n) -> int :
    i = 2;

    while i*i <= n :
        if n % i != 0 :
            i += 1
        else :
            n = n/i
    return n

def main() -> None :
    print(solve(600851475143))

if __name__ == "__main__":
    main()
